### DISCUSSION TITLE
Post 'huge document' results in Error

### DISCUSSION THREAD

**Discussion by RvZEU** [ravendb/ravendb#17019] [Q&A] [answered]
Sometimes we push a base64encoded image into a document, this is temporary and in this case it was slightly bigger than 5 MB

But at saving the document we get this error from RavenDB:

"Type":"System.ArgumentNullException","Message":"Value cannot be null. (Parameter 'key')","Error":"System.ArgumentNullException: Value cannot be null. (Parameter 'key')\n   at System.ThrowHelper.ThrowArgumentNullException(String name)\n   at Raven.Server.NotificationCenter.Notifications.Details.HugeDocumentsDetails.Update(String id, Int32 size) in C:\\Builds\\RavenDB-Stable-5.4\\54038\\src\\Raven.Server\\NotificationCenter\\Notifications\\Details\\HugeDocumentsDetails.cs:line 24\n   at Raven.Server.Documents.HugeDocuments.AddHint(String id, Int32 size) in C:\\Builds\\RavenDB-Stable-5.4\\54038\\src\\Raven.Server\\Documents\\HugeDocuments.cs:line 75\n  

We currently 'resolved' this issue by setting a bigger value for setting: PerformanceHints.Documents.HugeDocumentSizeInMb

To prevent future errors, is there we can do before posting this data or is this an issue in RavenDB itself? Version 5.4.103 is used

**Comment by garayx** [comment]
Hi

Can you share the code that you use for adding the document & adding the attachment ?

In any case the issue is with the notification regarding huge documents, not adding of the document (or attachment) itself.

**Comment by RvZEU** [comment]
Hi,

it was via a REST post call, the body was this, i removed the image data.

The image was not added as attachment but as a value in this post data

`{"Commands":[{"Type":"PUT","ChangeVector":null,"Id":null,"Document":{"entity":"products","entityId":"W108","importData":["W108","W108-liggend.png","W108-staand.png"],"newData":{"id":"HZMonFNq_24","externalId":"HZMonFNq_24","webshopId":24,"sku":"W108","name":"Wandpaneel W108","price":53.36,"specialPrice":null,"createdAt":"2023-06-29 12:01:31","updatedAt":"2023-08-08 10:26:32","productLinks":[{"sku":"W108","link_type":"calculator","linked_product_sku":"FDP500","linked_product_type":"simple","position":1,"extension_attributes":{"contents":2}},{"sku":"W108","link_type":"calculator","linked_product_sku":"FDP700","linked_product_type":"simple","position":2,"extension_attributes":{"contents":3}},{"sku":"W108","link_type":"calculator","linked_product_sku":"FX200","linked_product_type":"simple","position":3,"extension_attributes":{"contents":4}}],"tierPrices":[],"websites":[1],"stores":[{"id":"Stores\/HZMonFNq_1","name":"STYQX","websiteId":1,"externalId":1,"code":"default","connections":"Connections\/648987d826d068.43491085","websiteName":"STYQX"}],"storeNames":null,"description":"","shortDescription":null,"status":1,"visibility":4,"typeId":"simple","attributeSetId":22,"metaDescription":"Wandpaneel W108 Zigzag kopen? \u2713 Direct leverbaar \u2713 Montage in NL en BE \u2713 Sample service \u2713 Offici\u00eble Orac Decor\u00ae dealer \u2713 10 jaar ervaring","metaKeyword":"orac, decor, wall, covering, wandbekleding, muurbekleding, zigzag, wanddecoratie, W108, wandpaneel, wandpanelen,","metaTitle":"STYQX - Wandpaneel W108, unieke en hippe wandbekleding","weight":0,"collection":"Products","allowedToTranslate":false,"allowedToTranslateInTrial":false,"approvedTranslation":false,"syncedWebshop":true,"mediaGalleryEntries":[{"id":null,"media_type":"image","label":"","position":1,"disabled":false,"types":["image","small_image","thumbnail"],"file":"W108-liggend.png","content":{"base64_encoded_data":"","type":"image\/png","name":"W108-staand.png"}}],"attributes":{"image":"no_selection","url_key":"wandpaneel-w108","gift_message_available":"0","manufacturer":"6","styqx_free_shipping":"0","price_calculator_contents":"2.000000","sample":"1","small_image":"no_selection","options_container":"container2","lengte":"200\u200b,0 cm","price_calculator_unit":"15","sample_price":"0.820000","thumbnail":"no_selection","breedte":"25\u200b,0 cm","price_calculator_package_type":"18","swatch_image":"no_selection","dikte":"1\u200b,8 cm","materiaal":"Purotouch\u00ae (PU)","afwerking":"Reeds voorzien van witte primer","tax_class_id":"2","msrp_display_actual_price_type":"0","overschilderbaar":"Met lak of muurverf","waterbestendig":"Bestand tegen de meeste algemene oplosmiddelen en water","hittebestendig":"Tot max. 70 \u00b0C","termiet_resistent":"Ongevoelig voor ongedierte","ondergrond":"Vlak\u200b, droog \u200b, stof- en vetvrij","acclimatisatie":"Minimaal 24 uur voor de montage op kamertemperatuur","garantie":"Enkel met het gebruik van de aanbevolen DecoFix lijmen","ean":"5414433018160","akoestisch_effect":"Absorbeert tot 25% van een gesprek","mst_search_weight":"0"},"categoryIds":[{"position":0,"category_id":"6"},{"position":1,"category_id":"15"}],"options":[],"buckets":[],"translationStatus":null,"categories":[],"translations":[{"id":1}],"contentJobs":[],"dateSynced":"2023-08-08 12:26:32","editors":null,"syncMessage":"","syncStatus":"synced","lastChangedBy":"API","hasExternalChanges":false,"originalData":[],"needssync":null,"searchreplaces":null,"stockItem":[],"parentSkus":[],"childSkus":[],"extensionAttributes":{"website_ids":[1],"category_links":[{"position":0,"category_id":"6"},{"position":1,"category_id":"15"}]},"error":false,"isUpdated":1,"synchronisations":{"translation":[],"content_enrichment":[{"id":"64c0d75775ea50.26584709","dateSynced":"2023-08-02 21:08:54","syncStatus":"synced","syncMessage":""}],"import":[{"id":"64980d9d6a44f0.02464278","dateSynced":"2023-08-08 12:26:32","syncStatus":"synced","syncMessage":""}]}},"oldData":{"translationId":null,"translationCollection":null,"metaData":null,"id":"HZMonFNq_24","externalId":"HZMonFNq_24","webshopId":24,"sku":"W108","name":"Wandpaneel W108","price":53.36,"specialPrice":null,"createdAt":"2023-06-29 12:01:31","updatedAt":"2023-08-08 10:26:32","productLinks":[{"sku":"W108","link_type":"calculator","linked_product_sku":"FDP500","linked_product_type":"simple","position":1,"extension_attributes":{"contents":2}},{"sku":"W108","link_type":"calculator","linked_product_sku":"FDP700","linked_product_type":"simple","position":2,"extension_attributes":{"contents":3}},{"sku":"W108","link_type":"calculator","linked_product_sku":"FX200","linked_product_type":"simple","position":3,"extension_attributes":{"contents":4}}],"tierPrices":[],"websites":[1],"stores":[{"id":"Stores\/HZMonFNq_1","name":"STYQX","websiteId":1,"externalId":1,"code":"default","connections":"Connections\/648987d826d068.43491085","websiteName":"STYQX"}],"storeNames":null,"description":"","shortDescription":null,"status":1,"visibility":4,"typeId":"simple","attributeSetId":22,"metaDescription":"Wandpaneel W108 Zigzag kopen? \u2713 Direct leverbaar \u2713 Montage in NL en BE \u2713 Sample service \u2713 Offici\u00eble Orac Decor\u00ae dealer \u2713 10 jaar ervaring","metaKeyword":"orac, decor, wall, covering, wandbekleding, muurbekleding, zigzag, wanddecoratie, W108, wandpaneel, wandpanelen,","metaTitle":"STYQX - Wandpaneel W108, unieke en hippe wandbekleding","weight":0,"collection":"Products","allowedToTranslate":false,"allowedToTranslateInTrial":false,"approvedTranslation":false,"syncedWebshop":true,"mediaGalleryEntries":[],"attributes":{"image":"no_selection","url_key":"wandpaneel-w108","gift_message_available":"0","manufacturer":"6","styqx_free_shipping":"0","price_calculator_contents":"2.000000","sample":"1","small_image":"no_selection","options_container":"container2","lengte":"200\u200b,0 cm","price_calculator_unit":"15","sample_price":"0.820000","thumbnail":"no_selection","breedte":"25\u200b,0 cm","price_calculator_package_type":"18","swatch_image":"no_selection","dikte":"1\u200b,8 cm","materiaal":"Purotouch\u00ae (PU)","afwerking":"Reeds voorzien van witte primer","tax_class_id":"2","msrp_display_actual_price_type":"0","overschilderbaar":"Met lak of muurverf","waterbestendig":"Bestand tegen de meeste algemene oplosmiddelen en water","hittebestendig":"Tot max. 70 \u00b0C","termiet_resistent":"Ongevoelig voor ongedierte","ondergrond":"Vlak\u200b, droog \u200b, stof- en vetvrij","acclimatisatie":"Minimaal 24 uur voor de montage op kamertemperatuur","garantie":"Enkel met het gebruik van de aanbevolen DecoFix lijmen","ean":"5414433018160","akoestisch_effect":"Absorbeert tot 25% van een gesprek","mst_search_weight":"0"},"categoryIds":[{"position":0,"category_id":"6"},{"position":1,"category_id":"15"}],"options":[],"buckets":[],"translationStatus":null,"categories":[],"translations":[{"id":1}],"contentJobs":[],"dateSynced":"2023-08-08 12:26:32","editors":null,"syncMessage":"","syncStatus":"synced","lastChangedBy":"API","hasExternalChanges":false,"originalData":[],"needssync":null,"searchreplaces":null,"stockItem":[],"parentSkus":[],"childSkus":[],"extensionAttributes":{"website_ids":[1],"category_links":[{"position":0,"category_id":"6"},{"position":1,"category_id":"15"}]},"error":false,"isUpdated":1,"baseElements":["stockItem","attributeSetId"],"synchronisations":{"translation":[],"content_enrichment":[{"id":"64c0d75775ea50.26584709","dateSynced":"2023-08-02 21:08:54","syncStatus":"synced","syncMessage":""}],"import":[{"id":"64980d9d6a44f0.02464278","dateSynced":"2023-08-08 12:26:32","syncStatus":"synced","syncMessage":""}]}},"actionType":"update","importStatus":"pending","scheduleAt":false,"store":"Stores\/HZMonFNq_0","jobId":"E1xYkZ3Tp2","importId":"64aec142d6cf23.58688893","mappedFields":[{"csvheader":"0","mappedField":"sku"},{"csvheader":"1","mappedField":"main_image"},{"csvheader":"2","mappedField":"additional_image"}],"columns":{"main_image":{"current":[],"new":"W108-liggend.png"},"additional_image":{"current":"","new":"W108-staand.png"}},"messages":[],"@metadata":{"@collection":"ImportQueue"}}}]}`

And because of this issue with the notification the document was not created.

**Comment by ayende** [comment]
Aside from the actual error, why aren't you using an attachment for this?

**Comment by RvZEU** [comment]
Because a background process sends this data 1 to 1 to an API endpoint. No need for data manipulation on that side.

When it is send the image data will be unset.



Verzonden vanuit Outlook voor iOS<https://aka.ms/o0ukef>
________________________________
Van: Ayende Rahien ***@***.***>
Verzonden: Thursday, August 10, 2023 11:12:10 PM
Aan: ravendb/ravendb ***@***.***>
CC: Richard ***@***.***>; Author ***@***.***>
Onderwerp: Re: [ravendb/ravendb] Post 'huge document' results in Error (Discussion #17019)


Aside from the actual error, why aren't you using an attachment for this?

**Reply by ayende** [reply]
It would be better all around to do that as an attachment.

**Comment by grisha-kotler** [comment, marked as answer]
You need to send `"Id": ""` instead of `"Id": null`.
https://issues.hibernatingrhinos.com/issue/RavenDB-21074/NRE-in-Huge-Documents-notification
